### ISSUE TITLE
@Nullable in backend methods

### ISSUE THREAD

**Issue by dae** [ankidroid/Anki-Android-Backend#18]
I noticed in your recent docs that the backend methods are declared with nullable args, eg:

https://github.com/david-allison-1/Anki-Android-Backend/commit/799600f0d487a31137b553cbfb6e90f1a626c79d#diff-427e3c3bcf0309f9e862bb317502e4fc9cb32655a58df4ec9332ed08b36e88c3R79

Was there a particular reason for taking that approach? It is valid protobuf not to set them, and the backend shouldn't fall over if they are not set, but in most cases arguments are expected to be set to something, and a null argument is probably indicative of a bug in the calling code.  The way the Python code does it is to have all arguments of the autogenerated methods be required values, and it then becomes the responsibility of the wrapper code to explicitly provide default values for calls where it makes sense to - perhaps such an approach would work in Java too?

Not a high priority, just wanted to bring it up. And congratulations on getting to this point - it looks like you've put in a mammoth effort.

**Comment by david-allison-1** [comment]
TL;DR: Agreed. Will remove `@Nullable`, but won't switch to `@NonNull` in the interface code, 

I'll leave `@NonNull` as a decision for `DroidBackend` to take, as (from your comment) nullability is convention-based, rather than defined in the protos, and I'd rather leave `RustBackendImpl/RustBackend` auto-generated without business logic.

----


The autogenerated Java code for `Backend.SearchCardsIn.Builder.setSearch`  (via Google's protobuf-generation code) throws if a null argument is supplied.

```java
      /**
       * <code>string search = 1;</code>
       */
      public Builder setSearch(
          java.lang.String value) {
        if (value == null) {
    throw new NullPointerException();
  }
  
        search_ = value;
        onChanged();
        return this;
      }
```

----

I can easily remove the `@Nullable` from the above code as I'm in control of generation.

All of the Java access to the rust is via the `(Rust)DroidBackend` adapter, so setting default values for the method calls is possible, as is defining nullability.

**Comment by dae** [comment]
(can be unpinned)
